lifestuff/src/http_utils.rs

```diff
@@ -0,0 +1,93 @@
+// HTTP utility functions for API operations
+
+use url::Url;
+
+/// Check if a parsed URL host is localhost (localhost or 127.0.0.1)
+fn is_localhost_host(url: &Url) -> bool {
+    url.host_str()
+        .map(|h| h == "localhost" || h == "127.0.0.1")
+        .unwrap_or(false)
+}
+
+/// Check if a URL string points to localhost (localhost or 127.0.0.1)
+pub(crate) fn is_localhost_url(base_url: &str) -> bool {
```

Comment:
in ipv6 it could also be a different string.

Perhaps the best, true way to do this is to actaul do the resultion of the url and check if it matches

Reply:
for my usage it's probably okay to be honest. I'm not using ipv6 and the localhost thing is only to check if i'm running the endpoint locally on my laptop or if i'm actually hitting my production upstream endpoint. makes development much easier